extensions/emmet/src/abbreviationActions.ts

```diff
@@ -436,14 +443,20 @@ function expandAbbr(input: ExpandAbbreviationInput): string | undefined {
 		let expandedText = helper.expandAbbreviation(input.abbreviation, expandOptions);
 
 		if (input.textToWrap) {
+			// Fetch innermost element in the expanded abbreviation
```

Comment:
Shouldnt we do the search for block elements only if textToWrap was a single line?

`if (input.textToWrap.length === 1 && input.rangeToReplace.isSingleLine)`